site/app/controllers/admin/ReportController.php

```diff
@@ -367,6 +379,26 @@ private function saveUserToFile(string $base_path, User $user, array $ggs, LateD
         file_put_contents(FileUtils::joinPaths($base_path, $user->getId() . '_summary.json'), FileUtils::encodeJson($user_data));
     }
 
+    /**
+     * Generates a summary of all polls over a semester if polling is enabled
+     * @param string $base_path the base path to store the report
+     */
+    private function generatePollSummaryInternal($base_path) {
+        $polls = $this->core->getQueries()->getPolls();
+        $polls_data = [];
+        foreach ($polls as $poll) {
+            $poll_data = [];
+            $poll_data["name"] = $poll->getName();
+            $student_response_data = [];
+            foreach ($poll->getUserResponses() as $student => $response) {
+                $student_response_data[$student] = $poll->isCorrect($response);
+            }
+            $poll_data["responses"] = $student_response_data;
+            $polls_data[] = $poll_data;
```

Comment:
Very slight micro optimization, but we can save on the memory steps of creating a new variable and assigning data to it and instead just push the entire thing to `$polls_data` once we're ready:

```suggestion
            $student_response_data = [];
            foreach ($poll->getUserResponses() as $student => $response) {
                $student_response_data[$student] = $poll->isCorrect($response);
            }
            $poll_data["responses"] = $student_response_data;
            $polls_data[] = [
                'name' => $poll->getName(),
                'responses' => $student_response_data,
            ];
```